guide/src/class.md

```diff
@@ -41,20 +39,59 @@ Because Python objects are freely shared between threads by the Python interpret
 
 The above example generates implementations for [`PyTypeInfo`], [`PyTypeObject`], and [`PyClass`] for `MyClass` and `MyEnum`. To see these generated implementations, refer to the [implementation details](#implementation-details) at the end of this chapter.
 
+## Constructor
+
+By default it is not possible to create an instance of a custom class from Python code.
+To declare a constructor, you need to define a method and annotate it with the `#[new]`
+attribute. Only Python's `__new__` method can be specified, `__init__` is not available.
```

Comment:
(I know this is moved from lower down...)

I would love for us to think about whether we could / should support `__init__` in the future. Think it's worth making an issue and linking it here for interested users to join discussion?

Reply:
Is there an use case for it?

Given that Rust doesn't really do partial initialization (there's no clever constructor stuff, everything just bottoms out at creating a struct and initializing all its fields at once), it doesn't map well to that either.

I think it's better to not have an issue for now. We can discuss it on an issue if/when someone makes one.